src/utils/zcl_abapgit_log.clas.abap

```diff
@@ -37,6 +47,36 @@ ENDCLASS.
 
 CLASS zcl_abapgit_log IMPLEMENTATION.
 
+  METHOD constructor.
+
+    me->zif_abapgit_log~set_title( iv_title = iv_title ).
+
+  ENDMETHOD.
+
+  METHOD get_log.
+
+    CREATE OBJECT rv_result TYPE zcl_abapgit_log EXPORTING iv_title = iv_log_title.
+
+  ENDMETHOD.
+
+  METHOD get_repo_log.
+
+    DATA: ls_repo_log TYPE ty_repo_log_instance.
+    FIELD-SYMBOLS: <ls_repo_log> TYPE ty_repo_log_instance.
+
+    READ TABLE gt_repo_logs ASSIGNING <ls_repo_log> WITH KEY repo_key = iv_repo_key.
+    IF sy-subrc <> 0.
+      ls_repo_log-repo_key = iv_repo_key.
+      CREATE OBJECT ls_repo_log-instance TYPE zcl_abapgit_log EXPORTING iv_title = iv_log_title.
+      INSERT ls_repo_log INTO TABLE gt_repo_logs ASSIGNING <ls_repo_log>.
+    ELSEIF iv_log_title IS SUPPLIED.
+      <ls_repo_log>-instance->set_title( iv_log_title ).
```

Comment:
Mmmm ... singleton instance ? Could it already have some content ? and the the title redefined ? and what happens to the content ? Not really clear.